pages/id/[form]/settings.tsx

```diff
@@ -91,25 +91,13 @@ const FormSettings = (props: FormSettingsProps): React.ReactElement => {
 };
 
 export const getServerSideProps = requireAuthentication(async (context) => {
-  const formID = context && context.params && context.params.form ? context.params.form : undefined;
+  const formID = context?.params?.form ? parseInt(context.params.form as string) : undefined;
 
-  if (formID) {
+  if (formID === formID) {
```

Comment:
While true that `NaN !== NaN` this function will still permit a `undefined === undefined` which does resolve to true.  For simplicity I might refactor to something like:
```js
const formID = parseInt(context?.params?.form as string);
if (!isNaN(formID)){
...
}
```
Passing an `undefined` to `parseInt()` returns a NaN as well.

Reply:
Very good catch. Silly mistake on my part thank you

Reply:
TIL `isNaN` returns true for an input of undefined